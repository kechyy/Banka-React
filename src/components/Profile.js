import React, {Component} from 'react';
import { connect } from 'react-redux';
import '../assets/css/main.css';
import { profileAction } from '../store/modules/userProfile/actions' ;
import MainHeader from './MainHeader';
import MainFooter from './MainFooter';
import SideBar from './SideBar';

class Profile extends Component {
  constructor() {
    super();
    this.state = { page: 'profile', title: 'Profile Details', status:true, topMenu:true }
  }
  componentDidMount() {
    const { profileAction } = this.props;
    document.body.className = 'main-body';
    profileAction();
  }
  render() {
    return (
      <div>
        <MainHeader />
        <main>
          <div className="wrapper ">
            <SideBar />
            <div className="right">
              <div className="row">
                <div className="col-12-lg mainContent">
                  <h2 id="contentMainTitle" >User Profile Page</h2>
                  <header className="mainTitle">
                    <div className="row ">
                      <div className="col-12-md">
                        <h3 id="contentSubTitle" >{this.state.title}</h3>
                      </div>
                    </div>
                  </header>
                  <aside className="rightSideBar">
                    <div className="row" id="userMain">
                    <div style={{'width':'100%'}} >
                      <div className="col-8-md acctDetails">
                          <table>
                              <tbody>
                                  <tr>
                                      <td><strong> First Name</strong></td>
                                      <td><strong>{this.props.userProfile.firstname}</strong></td>
                                  </tr>
                                  <tr>
                                      <td><strong>LastName</strong> </td>
                                      <td>{this.props.userProfile.lastname}</td>
                                  </tr>
                                  <tr>
                                      <td><strong>Email Address</strong> </td>
                                      <td>{this.props.userProfile.email}</td>
                                  </tr>
                                  <tr>
                                      <td><strong>User type</strong> </td>
                                      <td>{this.props.userProfile.usertype}</td>
                                  </tr>
                                  <tr>
                                      <td><strong>Status</strong> </td>
                                      <td>{this.props.userProfile.user_email_status}</td>
                                  </tr>
                              </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </aside>
                </div>
              </div> 
            </div>
          </div>
        </main>
       <MainFooter />
       </div>
    )
  }
} 
const mapStateToProps = state => ({
  userProfile: state.profileReducer.profile
});

const mapDispatchToProps = {
  profileAction
}
export default connect(mapStateToProps, mapDispatchToProps)(Profile);