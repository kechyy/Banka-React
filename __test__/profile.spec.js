import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount, shallow } from './enzyme';
import Profile from '../src/components/Profile';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { BrowserRouter, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';


const mockStore = configureMockStore([thunk]);
let store;
const initialSate = {
  profile: {},
  status: 'rest'
};

const props = {
  isAuthenticated: true,
  profileAction: jest.fn(),
  userProfile: jest.fn()
};

const renderWithEnzymes = state => {
  store = mockStore({
    authReducer: {
      error: null,
      isAuthenticated: false,
      user: {},
      status: 'rest'
    },
    profileReducer: {
      profile: {},
      status: 'rest'
    },
    accountReducer: {
      account: [],
      error: null,
      status: 'rest'
    },
    Profile: {
      ...state
    }
  });
  return mount(
    <Provider store={store}>
      <Router>
        <Profile {...props} />
      </Router>
    </Provider>
  );
};


describe('Profile', () => {
  afterEach(() => {
    jest.clearAllMocks();
    store = null;
  });

  it('should have 1 table on profile page', () => {
    const wrapper = renderWithEnzymes();

    expect(wrapper.find('table').length).toBe(1);
   
  });

  it('should render Profile Component', () => {
    const wrapper = renderWithEnzymes();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});