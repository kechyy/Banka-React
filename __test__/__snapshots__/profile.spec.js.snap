// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render Profile Component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Profile)
        isAuthenticated={true}
        profileAction={[MockFunction]}
        userProfile={[MockFunction]}
      >
        <Profile
          isAuthenticated={true}
          profileAction={[Function]}
          userProfile={Object {}}
        >
          <div>
            <Connect(MainHeader)>
              <MainHeader
                logoutAction={[Function]}
                profileAction={[Function]}
              >
                <header
                  className="mainHeader"
                >
                  <div
                    className="wrapper"
                  >
                    <div
                      className="container topHeader"
                    >
                      <div
                        className="row"
                        style={
                          Object {
                            "justifyContent": "flex-end",
                          }
                        }
                      >
                        <Link
                          className="logout btn btn-pink"
                          id="logout"
                          onClick={[Function]}
                          to="/signin"
                        >
                          <a
                            className="logout btn btn-pink"
                            href="/signin"
                            id="logout"
                            onClick={[Function]}
                          >
                            LOGOUT
                          </a>
                        </Link>
                        <button
                          className="topMenu btn btn-red"
                          onClick={[Function]}
                          style={
                            Object {
                              "padding": "10px",
                            }
                          }
                        >
                          Menu
                        </button>
                        <ul
                          className="dropDown"
                          style={
                            Object {
                              "display": "none",
                            }
                          }
                        >
                          <li
                            className="userProfiles"
                          >
                            <a
                              data-page="profile"
                              data-title="Profile Details"
                            >
                              Profile
                            </a>
                          </li>
                          <li
                            className="accountForms"
                          >
                            <span
                              className="fa fa-book"
                            />
                            <a
                              data-page="account"
                              data-title="Create Account"
                            >
                              Create Bank Account
                            </a>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                </header>
              </MainHeader>
            </Connect(MainHeader)>
            <main>
              <div
                className="wrapper "
              >
                <Connect(SideBar)>
                  <SideBar
                    account={Array []}
                    getTransactionAction={[Function]}
                    userProfile={Object {}}
                  >
                    <div
                      className="left mainSideBar"
                    >
                      <div
                        className="row"
                      >
                        <div
                          className="col-8-lg sideBar"
                        >
                          <aside>
                            <div>
                              <img
                                alt=""
                                className="logo"
                                src="../images/logo2.png"
                              />
                            </div>
                            <div
                              className="sideBarTitle"
                            >
                              Client User
                            </div>
                            <ul
                              className="sideMenu"
                            >
                              <li
                                className="userProfiles"
                              >
                                -  
                                <Link
                                  data-title="Profile Details"
                                  to="/user-profile"
                                >
                                  <a
                                    data-title="Profile Details"
                                    href="/user-profile"
                                    onClick={[Function]}
                                  >
                                    Profile
                                  </a>
                                </Link>
                              </li>
                              <li
                                className="accountForms"
                              >
                                <span
                                  className="fa fa-book"
                                />
                                <Link
                                  data-title="Create Account"
                                  to="/create-account"
                                >
                                  <a
                                    data-title="Create Account"
                                    href="/create-account"
                                    onClick={[Function]}
                                  >
                                    Create Bank Account
                                  </a>
                                </Link>
                              </li>
                              <li
                                className="transact"
                              >
                                -  
                                <a
                                  onClick={[Function]}
                                >
                                  Transaction History
                                </a>
                                <ul
                                  className="sideBarDropDown"
                                  style={
                                    Object {
                                      "display": "none",
                                    }
                                  }
                                />
                              </li>
                            </ul>
                          </aside>
                        </div>
                      </div>
                    </div>
                  </SideBar>
                </Connect(SideBar)>
                <div
                  className="right"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-12-lg mainContent"
                    >
                      <h2
                        id="contentMainTitle"
                      >
                        User Profile Page
                      </h2>
                      <header
                        className="mainTitle"
                      >
                        <div
                          className="row "
                        >
                          <div
                            className="col-12-md"
                          >
                            <h3
                              id="contentSubTitle"
                            >
                              Profile Details
                            </h3>
                          </div>
                        </div>
                      </header>
                      <aside
                        className="rightSideBar"
                      >
                        <div
                          className="row"
                          id="userMain"
                        >
                          <div
                            style={
                              Object {
                                "width": "100%",
                              }
                            }
                          >
                            <div
                              className="col-8-md acctDetails"
                            >
                              <table>
                                <tbody>
                                  <tr>
                                    <td>
                                      <strong>
                                         First Name
                                      </strong>
                                    </td>
                                    <td>
                                      <strong />
                                    </td>
                                  </tr>
                                  <tr>
                                    <td>
                                      <strong>
                                        LastName
                                      </strong>
                                       
                                    </td>
                                    <td />
                                  </tr>
                                  <tr>
                                    <td>
                                      <strong>
                                        Email Address
                                      </strong>
                                       
                                    </td>
                                    <td />
                                  </tr>
                                  <tr>
                                    <td>
                                      <strong>
                                        User type
                                      </strong>
                                       
                                    </td>
                                    <td />
                                  </tr>
                                  <tr>
                                    <td>
                                      <strong>
                                        Status
                                      </strong>
                                       
                                    </td>
                                    <td />
                                  </tr>
                                </tbody>
                              </table>
                            </div>
                          </div>
                        </div>
                      </aside>
                    </div>
                  </div>
                </div>
              </div>
            </main>
            <MainFooter>
              <footer
                className="footer"
              >
                <p>
                  Copyright © Kecyy Banka Application all 2019 rights reserved
                </p>
              </footer>
            </MainFooter>
          </div>
        </Profile>
      </Connect(Profile)>
    </Router>
  </BrowserRouter>
</Provider>
`;
